modules/meta/arrow-meta-prototype/idea-plugin/src/main/kotlin/arrow/meta/plugin/idea/phases/resolve/DiagnosticSuppressor.kt

```diff
@@ -0,0 +1,58 @@
+package arrow.meta.plugin.idea.phases.resolve
+
+import org.jetbrains.kotlin.cfg.ClassMissingCase
+import org.jetbrains.kotlin.cfg.WhenMissingCase
+import org.jetbrains.kotlin.descriptors.VariableDescriptor
+import org.jetbrains.kotlin.diagnostics.Diagnostic
+import org.jetbrains.kotlin.diagnostics.DiagnosticWithParameters1
+import org.jetbrains.kotlin.diagnostics.DiagnosticWithParameters2
+import org.jetbrains.kotlin.diagnostics.Errors
+import org.jetbrains.kotlin.psi.KtElement
+import org.jetbrains.kotlin.psi.KtParameter
+import org.jetbrains.kotlin.psi.KtWhenExpression
+import org.jetbrains.kotlin.types.KotlinType
+import org.jetbrains.kotlin.types.checker.KotlinTypeChecker
+import org.jetbrains.kotlin.utils.addToStdlib.safeAs
+
+open class DiagnosticSuppressor : org.jetbrains.kotlin.resolve.diagnostics.DiagnosticSuppressor {
```

Comment:
Is there a reason why this has to be an open class?

Reply:
No reason